pkg/util/net/client.go

```diff
@@ -31,7 +31,9 @@ func GetTransport(extras map[string]string, rt http.RoundTripper) *headerTranspo
 // on every request.
 func NewHeaderClient(extras map[string]string, rt http.RoundTripper) *http.Client {
 	if len(extras) == 0 {
-		return http.DefaultClient
+		return &http.Client{
```

Comment:
This seems like an unrelated bug fix. Can you put it in its own commit and discuss the triggers/impact of the bug?